src/types/secret_key_repr.rs

```diff
@@ -21,24 +23,78 @@ pub enum SecretKeyRepr {
     EdDSA(EdDSASecretKey),
 }
 
-/// Secret key for ECDH with Curve25519, the only combination we currently support.
+/// Secret key for ECDH with Curve25519 or P-521, the only combinations we currently support.
 #[derive(Clone, PartialEq, Eq, Zeroize, ZeroizeOnDrop)]
-pub struct ECDHSecretKey {
-    /// The secret point.
-    pub secret: [u8; 32],
-    pub hash: HashAlgorithm,
-    pub oid: Vec<u8>,
-    pub alg_sym: SymmetricKeyAlgorithm,
+#[non_exhaustive]
+pub enum ECDHSecretKey {
+    Cv25519 {
+        /// The secret point.
+        secret: [u8; 32],
+        hash: HashAlgorithm,
+        oid: Vec<u8>,
+        alg_sym: SymmetricKeyAlgorithm,
+    },
+    P521 {
+        /// The secret point.
+        secret: [u8; <NistP521 as Curve>::FieldBytesSize::USIZE],
+        hash: HashAlgorithm,
+        oid: Vec<u8>,
+        alg_sym: SymmetricKeyAlgorithm,
+    },
+}
```

Comment:
Curious if something like this could work as well, with possibly a smaller delta:

```suggestion
#[non_exhaustive]
pub struct ECDHSecretKey {
    /// The secret point.
    pub secret: ECDHSecretPoint,
    pub hash: HashAlgorithm,
    pub oid: Vec<u8>,
    pub alg_sym: SymmetricKeyAlgorithm,
}

/// The secret point.
#[non_exhaustive]
pub enum ECDHSecretPoint {
    Cv25519([u8; 32]),
    P521([u8; <NistP521 as Curve>::FieldBytesSize::USIZE]),
}
```